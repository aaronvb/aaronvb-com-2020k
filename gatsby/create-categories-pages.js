'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___category) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (category) => {
    const categorySlug = `/category/${_.kebabCase(category.fieldValue)}`;

    createPage({
      path: categorySlug,
      component: path.resolve('./src/templates/category-template.js'),
      context: {
        category: category.fieldValue,
      }
    });
  });
};
