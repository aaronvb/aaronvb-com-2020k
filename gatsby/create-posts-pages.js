'use strict';

const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  createPage({
    path: '/',
    component: path.resolve('./src/templates/index-template.js'),
  });
};
