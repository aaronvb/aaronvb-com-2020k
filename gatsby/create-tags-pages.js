'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___tags) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (tag) => {
    const tagSlug = `/tag/${_.kebabCase(tag.fieldValue)}`;

    createPage({
      path: tagSlug,
      component: path.resolve('./src/templates/tag-template.js'),
      context: {
        tag: tag.fieldValue,
      }
    });
  });
};
