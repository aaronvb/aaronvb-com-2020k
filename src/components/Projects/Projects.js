// @flow strict
import React from 'react';
import styles from './Projects.module.scss';

type Props = {
  body: string,
}

const Projects = ({ body }: Props) => (
  <div className={styles['projects']}>
    <div className={styles['projects__inner']}>
      <h4 className={styles['page__title']} href="#projects" id="projects">Projects</h4>
      <div className={styles['projects__body']} dangerouslySetInnerHTML={{ __html: body }} />
    </div>
  </div>
);

export default Projects;