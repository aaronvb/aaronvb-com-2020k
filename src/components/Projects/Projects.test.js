// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Projects from './Projects';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

describe('Projects', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  const props = {
    body: '<p>test</p>'
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Projects { ...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
