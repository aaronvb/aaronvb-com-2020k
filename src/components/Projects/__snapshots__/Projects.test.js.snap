// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders correctly 1`] = `
<div
  className="projects"
>
  <div
    className="projects__inner"
  >
    <h4
      className="page__title"
      href="#projects"
      id="projects"
    >
      Projects
    </h4>
    <div
      className="projects__body"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>test</p>",
        }
      }
    />
  </div>
</div>
`;
